server/src/main/java/io/deephaven/server/console/python/DebuggingInitializer.java

```diff
@@ -0,0 +1,45 @@
+package io.deephaven.server.console.python;
+
+import io.deephaven.configuration.Configuration;
+import io.deephaven.util.thread.ThreadInitializationFactory;
+import org.jpy.PyLib;
+import org.jpy.PyModule;
+import org.jpy.PyObject;
+
+import java.io.Closeable;
+
+/**
+ * If python is configured as the script language for this server, ensures that threads which may invoke python code
+ * will be able to be debugged. If python is disabled, this does nothing.
+ */
+public class DebuggingInitializer implements ThreadInitializationFactory {
+    @Override
+    public Runnable createInitializer(Runnable runnable) {
+        if (!"python".equals(Configuration.getInstance().getStringWithDefault("deephaven.console.type", null))) {
+            // python not enabled, don't accidentally start it
+            return runnable;
+        }
+        DeephavenModule py_deephaven = (DeephavenModule) PyModule.importModule("deephaven")
+                .createProxy(PyLib.CallableKind.FUNCTION, DeephavenModule.class);
+
+
+        return () -> {
+            // First call in to create a custom function that has the same name as the Java thread (plus a prefix)
+            PyObject runnableResult = py_deephaven.create_thread_entry(Thread.currentThread().getName());
+            // runnable.run();
+            // Invoke that function directly from Java, so that we have only this one initial frame
+            runnableResult.call("__call__", runnable);
```

Comment:
I think we could do this with a single method. The naming would probably change:

`py_deephaven.run_for_java_thread(Thread.currentThread().getName(), runnable)` or similar

Reply:
Potentially, and that's how the first implementation worked. However, if we wanted the top-most frame to have a specific name (so that you can tell what Java thread something was running on by looking at the py stack trace, which doesn't come with thread info), we would need to create a custom def to return in the call - the debugger at least loses the `__name__` and `__qualname__` properties that are assigned to rename defs. As such, we may need to return a pyobject so that it can be called again. 

And, if we just use the runnable.run() above (presently commented out), then we don't need to pass runnable at all, so the signature stays the same. In that case however it appears to the debugger that any pydevd.settrace() call in the web IDE actually pauses in the `create_change_list` function.

Reply:
Would this provide the same benefits?

```python
def run_for_java_thread(name, runnable):
    ...
    def JavaThread():
        runnable.run()
    JavaThread()
```

Reply:
* Changing the function's name (via declaring `JavaThread` not by name, but through eval) so the thread appears: technically yes, but by adding two frames instead of one, so the topmost frame isn't named in this way.
* Adding one py frame earlier in the stack than the current code: technically yes, but again, because it adds two frames instead of one, we might be making it harder to understand instead of easier

Remember, these run once per thread, when the thread is created, so on the order of 10ish times on startup and once when the first python command is executed from the web console, so paying the extra jni/py round trip is pretty cheap.